privval/socket_dialers_test.go

```diff
@@ -1,26 +1,49 @@
 package privval
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	"github.com/tendermint/tendermint/crypto/ed25519"
 	cmn "github.com/tendermint/tendermint/libs/common"
 )
 
+func getDialerTestCases(t *testing.T) []dialerTestCase {
+	tcpAddr := GetFreeLocalhostAddrPort()
+	unixFilePath, err := testUnixAddr()
+	require.NoError(t, err)
+	unixAddr := fmt.Sprintf("unix://%s", unixFilePath)
+
+	return []dialerTestCase{
+		{
+			addr:   tcpAddr,
+			dialer: DialTCPFn(tcpAddr, testTimeoutReadWrite, ed25519.GenPrivKey()),
+		},
+		{
+			addr:   unixAddr,
+			dialer: DialUnixFn(unixFilePath),
+		},
+	}
+}
+
 func TestIsConnTimeoutForFundamentalTimeouts(t *testing.T) {
 	// Generate a networking timeout
-	dialer := DialTCPFn(testFreeTCPAddr(t), time.Millisecond, ed25519.GenPrivKey())
+	tcpAddr := GetFreeLocalhostAddrPort()
+	dialer := DialTCPFn(tcpAddr, time.Millisecond, ed25519.GenPrivKey())
 	_, err := dialer()
 	assert.Error(t, err)
 	assert.True(t, IsConnTimeout(err))
 }
 
 func TestIsConnTimeoutForWrappedConnTimeouts(t *testing.T) {
-	dialer := DialTCPFn(testFreeTCPAddr(t), time.Millisecond, ed25519.GenPrivKey())
+	tcpAddr := GetFreeLocalhostAddrPort()
+	dialer := DialTCPFn(tcpAddr, time.Millisecond, ed25519.GenPrivKey())
 	_, err := dialer()
 	assert.Error(t, err)
-	err = cmn.ErrorWrap(ErrConnTimeout, err.Error())
+	err = cmn.ErrorWrap(ErrListenerTimeout, err.Error())
```

Comment:
```suggestion
	err = errors.Wrap(ErrListenerTimeout, err.Error())
```

Reply:
`cmn.ErrorWrap` and `error.Wrap` behave differently. When replaced the test fails as the wrapped error is not recognized as a timeout anymore.
Maybe we need to analyze if refactoring and replacing cmn.ErrorWrap makes sense to avoid possible bugs.

Reply:
Yeah, let's hold on this until next time

Reply:
I am worried that people may try to replace `cmn.ErrorWrap` with `errors.Wrap` expecting them to be equivalent, as per your suggestion. However, they are actually quite different.
Does it make sense that we still have both? or may we settle on only one of them?